gemrb/GUIScripts/GUISTORE.py

```diff
@@ -921,18 +948,73 @@ def SellPressed ():
 	Window = StoreShoppingWindow
 
 	pc = GemRB.GameGetSelectedPCSingle ()
-	RightCount = len (inventory_slots)
-	#no need to go reverse
-	for Slot in range (RightCount):
-		Flags = GemRB.IsValidStoreItem (pc, inventory_slots[Slot], ITEM_PC) & SHOP_SELECT
-		if Flags:
-			GemRB.ChangeStoreItem (pc, inventory_slots[Slot], SHOP_SELL)
+	if Bag:
+		RightCount = Bag['StoreItemCount']
+		#going backwards because removed items shift the slots
+		for Slot in range (RightCount-1, -1, -1):
+			Flags = GemRB.IsValidStoreItem (pc, Slot, ITEM_BAG)
+			if Flags & SHOP_SELECT:
+				#NOTE: what if the transaction fails?
+				GemRB.ChangeStoreItem (pc, Slot, SHOP_SELL)
+	else:
+		RightCount = len (inventory_slots)
+		#no need to go reverse
+		for Slot in range (RightCount):
+			Flags = GemRB.IsValidStoreItem (pc, inventory_slots[Slot], ITEM_PC) & SHOP_SELECT
+			if Flags:
+				GemRB.ChangeStoreItem (pc, inventory_slots[Slot], SHOP_SELL)
 
 	GemRB.GameSetPartyGold (GemRB.GameGetPartyGold ()+SellSum)
 	GemRB.PlaySound(DEF_SOLD)
 	UpdateStoreShoppingWindow ()
 	return
 
+def OpenBag ():
+	if Bag:
+		#nested containers are not openable in bg2,
+		#but double-click has another function here
+		OpenItemAmountWindow (STORE_BAG)
+		return
+	if Inventory:
+		return
+	pc = GemRB.GameGetSelectedPCSingle ()
+	RightIndex = GemRB.GetVar ("RightIndex")
+	Slot = GemRB.GetSlotItem (pc, inventory_slots[RightIndex])
+	Item = GemRB.GetItem (Slot['ItemResRef'])
+	if Item['Function'] & 4:
```

Comment:
Possible future improvement: add named constants for these bits to GUIDefines.py; it would also make it obvious this is checking for a container.